pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -395,6 +331,30 @@ func nameAndNamespace(serviceHostname, domain string) (name, namespace string) {
 	return serviceHostname, ""
 }
 
-func attrStringValue(value string) *mpb.Attributes_AttributeValue {
+func attrStringValue(value string) attribute {
 	return &mpb.Attributes_AttributeValue{Value: &mpb.Attributes_AttributeValue_StringValue{StringValue: value}}
 }
+
+func attrUID(node *model.Proxy) attribute {
+	return attrStringValue("kubernetes://" + node.ID)
+}
+
+func attrBoolValue(value bool) attribute {
+	return &mpb.Attributes_AttributeValue{Value: &mpb.Attributes_AttributeValue_BoolValue{BoolValue: value}}
+}
+
+func attrIntValue(value int64) attribute {
+	return &mpb.Attributes_AttributeValue{Value: &mpb.Attributes_AttributeValue_Int64Value{Int64Value: value}}
+}
+
+func attrIPValue(ip string) attribute {
+	return &mpb.Attributes_AttributeValue{Value: &mpb.Attributes_AttributeValue_BytesValue{BytesValue: net.ParseIP(ip)}}
```

Comment:
if not returning error, at least log if ip parses to nil. it's not clear what context this would be used from and whether ip can be assumed to be valid.

Reply:
OK. I just moved this code around from how it was before. Are you sure log spam is acceptable here?